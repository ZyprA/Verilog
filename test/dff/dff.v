module DFF(clk, d, q);
    input clk, d;
    output q;

    reg q;

    always @ (posedge clk)
        q <= d;

endmodule

`timescale 1ns / 1ns

module DFF_sim;
    reg clk, d;
    wire q;

    DFF dff0( clk, d, q);

        initial begin
        $dumpfile("dff.vcd");
        $dumpvars(0, DFF_sim);
    end

    initial begin
        clk = 0;
        forever begin
            #4 clk =~ clk;
        end
    end

    initial begin
        d = 0;
        #100 d = 1;
        #100 d = 0;
        #100 $finish;
    end

endmodule
