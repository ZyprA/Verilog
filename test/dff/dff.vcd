$date
	Sun Feb  2 16:18:36 2025
$end
$version
	Icarus Verilog
$end
$timescale
	1ns
$end
$scope module DFF_sim $end
$var wire 1 ! q $end
$var reg 1 " clk $end
$var reg 1 # d $end
$scope module dff0 $end
$var wire 1 " clk $end
$var wire 1 # d $end
$var reg 1 ! q $end
$upscope $end
$upscope $end
$enddefinitions $end
$comment Show the parameter values. $end
$dumpall
$end
#0
$dumpvars
0#
0"
x!
$end
#4
0!
1"
#8
0"
#12
1"
#16
0"
#20
1"
#24
0"
#28
1"
#32
0"
#36
1"
#40
0"
#44
1"
#48
0"
#52
1"
#56
0"
#60
1"
#64
0"
#68
1"
#72
0"
#76
1"
#80
0"
#84
1"
#88
0"
#92
1"
#96
0"
#100
1!
1"
1#
#104
0"
#108
1"
#112
0"
#116
1"
#120
0"
#124
1"
#128
0"
#132
1"
#136
0"
#140
1"
#144
0"
#148
1"
#152
0"
#156
1"
#160
0"
#164
1"
#168
0"
#172
1"
#176
0"
#180
1"
#184
0"
#188
1"
#192
0"
#196
1"
#200
0"
0#
#204
0!
1"
#208
0"
#212
1"
#216
0"
#220
1"
#224
0"
#228
1"
#232
0"
#236
1"
#240
0"
#244
1"
#248
0"
#252
1"
#256
0"
#260
1"
#264
0"
#268
1"
#272
0"
#276
1"
#280
0"
#284
1"
#288
0"
#292
1"
#296
0"
#300
1"
