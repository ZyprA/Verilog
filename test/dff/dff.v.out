#! /opt/homebrew/Cellar/icarus-verilog/12.0/bin/vvp
:ivl_version "12.0 (stable)";
:ivl_delay_selection "TYPICAL";
:vpi_time_precision - 9;
:vpi_module "/opt/homebrew/Cellar/icarus-verilog/12.0/lib/ivl/system.vpi";
:vpi_module "/opt/homebrew/Cellar/icarus-verilog/12.0/lib/ivl/vhdl_sys.vpi";
:vpi_module "/opt/homebrew/Cellar/icarus-verilog/12.0/lib/ivl/vhdl_textio.vpi";
:vpi_module "/opt/homebrew/Cellar/icarus-verilog/12.0/lib/ivl/v2005_math.vpi";
:vpi_module "/opt/homebrew/Cellar/icarus-verilog/12.0/lib/ivl/va_math.vpi";
S_0x14ee08270 .scope module, "DFF_sim" "DFF_sim" 2 14;
 .timescale -9 -9;
v0x14ee1c1b0_0 .var "clk", 0 0;
v0x14ee1c250_0 .var "d", 0 0;
v0x14ee1c300_0 .net "q", 0 0, v0x14ee1c0b0_0;  1 drivers
S_0x14ee07520 .scope module, "dff0" "DFF" 2 18, 2 1 0, S_0x14ee08270;
 .timescale 0 0;
    .port_info 0 /INPUT 1 "clk";
    .port_info 1 /INPUT 1 "d";
    .port_info 2 /OUTPUT 1 "q";
v0x14ee083e0_0 .net "clk", 0 0, v0x14ee1c1b0_0;  1 drivers
v0x14ee1c010_0 .net "d", 0 0, v0x14ee1c250_0;  1 drivers
v0x14ee1c0b0_0 .var "q", 0 0;
E_0x14ee07b00 .event posedge, v0x14ee083e0_0;
    .scope S_0x14ee07520;
T_0 ;
    %wait E_0x14ee07b00;
    %load/vec4 v0x14ee1c010_0;
    %assign/vec4 v0x14ee1c0b0_0, 0;
    %jmp T_0;
    .thread T_0;
    .scope S_0x14ee08270;
T_1 ;
    %vpi_call 2 21 "$dumpfile", "dff.vcd" {0 0 0};
    %vpi_call 2 22 "$dumpvars", 32'sb00000000000000000000000000000000, S_0x14ee08270 {0 0 0};
    %end;
    .thread T_1;
    .scope S_0x14ee08270;
T_2 ;
    %pushi/vec4 0, 0, 1;
    %store/vec4 v0x14ee1c1b0_0, 0, 1;
T_2.0 ;
    %delay 4, 0;
    %load/vec4 v0x14ee1c1b0_0;
    %inv;
    %store/vec4 v0x14ee1c1b0_0, 0, 1;
    %jmp T_2.0;
    %end;
    .thread T_2;
    .scope S_0x14ee08270;
T_3 ;
    %pushi/vec4 0, 0, 1;
    %store/vec4 v0x14ee1c250_0, 0, 1;
    %delay 100, 0;
    %pushi/vec4 1, 0, 1;
    %store/vec4 v0x14ee1c250_0, 0, 1;
    %delay 100, 0;
    %pushi/vec4 0, 0, 1;
    %store/vec4 v0x14ee1c250_0, 0, 1;
    %delay 100, 0;
    %vpi_call 2 36 "$finish" {0 0 0};
    %end;
    .thread T_3;
# The file index is used to find the file name in the following table.
:file_names 3;
    "N/A";
    "<interactive>";
    "dff.v";
